### PR TITLE
[Major update] Move to platform wide vector based QR code generation

### PR DESCRIPTION
**Pull request by sourobrata**
**Summary:**

This Pull request aims at completely moving away from raster-based QR code generation, which was previously the default, using HTML canvas. The new version will use SVG as the default format, which was previously only used for the .svg and .eps formats. PNG, JPG, PDF and EPS will now be generated from the source SVG.

**This addresses the following:**

- Maintainability of the core QR code library by removing the need to develop separately for SVG and canvas
- Consistency across all formats of QR codes generated by Beaconstac
- Easier deployments to Serverless lambda functions, by removing the dependency on a lambda layer for canvas

### CODE REVIEW

**Comment by sourobrata** [comment]
@aditya-beaconstac Lets start refactoring and cleaning up the code so that we can begin code reviews.

File: src/Util.ts [hunk 1]
```diff
@@ -59,8 +63,9 @@ export const loadImage = (src: string, imageServerURL?: string | undefined, imag
     }
     let image: any;
     if (isNode) {
-        image = new ImageCanvas();
+        // image = new ImageCanvas();
```
Comment (sourobrata) [inline src/Util.ts:66]: Redundant condition.

File: src/index.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ export class QRCodeBuilder {
     }
 
     public async build(format?: CanvasType): Promise<QRCode | never> {
-        this.config.canvasType = format ? format : CanvasType.PNG;
+        this.config.canvasType = format ? format : CanvasType.SVG;
```
Comment (sourobrata) [inline src/index.ts:37]: @aditya-beaconstac Can we cleanup the CanvasType enum?

**Comment by sourobrata** [comment]
@aditya-beaconstac Did we change the size of the input for the tests? I see a lot of these diffs.
Screenshot 2023-01-09 at 10 47 45 AM

File: src/tests/Common.test.ts [hunk 3]
```diff
@@ -1,5 +1,4 @@
 import { BCH, QRMath, Util, CanvasUtil } from '../Common';
-import { Image } from 'canvas';
 import { expect } from 'chai';
 import 'mocha';
```
Comment (sourobrata) [inline src/tests/Common.test.ts:3]: @aditya-beaconstac Can we drop this file?

**Comment by sourobrata** [comment]
@aditya-beaconstac There seems to be quite a few diffs in the output test files. Can you please look into it?

**Review by shreedhar-mobstac — APPROVED** [review]

**Pull request merged**